### DISCUSSION TITLE
WebREPL on ESP32-PICO

### DISCUSSION THREAD

**Discussion by TomLinz** [micropython/micropython#10978] [ESP32] [answered]
Trying to setup webREPL on my ESP32-PICO-KIT board, firmware esp32-20220618-v1.19.1.
I use Thonny 4.0.2 as my IDE. As the client I have downloaded from GitHub the "webrepl-master"
 -package and from there I open the webrepl.html file on Firefox.

After reading a number of various tutorials from the net, and setting up everything,
I am still without final success.

I get to asking for my psw, then greeting "WebREPL connected" and the ">>>" prompt 
appears in the client window, but when entering Python commands nothing happens. 
Not even echo while I type. 

However, I can get files from the ESP using the button "Get from Device". So there 
definitely is a WLAN connection between my PC and the ESP32.

Also, strangely, things work OK if I use Thonny as the client, by changing the drop down setting 
"Port or WebREPL" to <WebREPL>, reachable via Run -> Configure interpreter.

But as said, not working from the browser page. 

What am I missing here, please?

**Comment by sethstenzel** [comment, marked as answer]
Are you by chance connected to webrepl and over serial repl at the same time?
Sometimes when I do this, I can get some glitchy behavior where things typed don't show up, are latent, or bug out on either the web repl, or the serial connection repl.

**Comment by TomLinz** [comment]
Hi Seth, thanks for your response. I had a same feeling that Thonny might somehow interfere but did not have time to do a definite test. Today I totally disconnected the ESP32 board from my PC and fed it from a mobile phone charger. Amazingly, webREPL came to life.

Somewhere, I have seen remarks that webREPL is an "experimental" feature. Beware.
